src/process.c

```diff
@@ -347,7 +347,7 @@ cc_run_hook(struct oci_cfg_hook* hook, const gchar* state,
 		close_if_set (stdin_pipe[0]);
 		close_if_set (pipe_child_error[1]);
 		close_if_set (pipe_parent_error[0]);
-		abort();
+		exit(EXIT_FAILURE);
```

Comment:
I would prefer to call exit() with the reason so there is more discrimination on why the hook failed, rather than just "it failed". Of course errno can be altered by things like the close_if_set() calls, so really need to create a new variable and assign it in all the failure paths to capture the first failure.

So for example put an assignment before line 338 - the call to g_critical.

The idea is sound - just give a little more info.